routing/router.go

```diff
@@ -0,0 +1,93 @@
+package routing
```

Comment:
Czym jest dla Ciebie routing i jak myślisz co on powinien robić? Luźna dyskusja, a później skomentuje ten pakiet :)

Reply:
Routing jest wg. mnie nasłuchiwaczem połączeń z/do danego endpointa.
Moim zdaniem odpowiada za przekierowywanie przychodzącego zapytania do odpowiedniego handlera i ewentualnej autoryzacji. Stąd pomysł taki, aby w tablicy routingów dać pole typu bool, które by sprawdzało, czy dany element wymaga autoryzacji, ale nie wiedziałem, jak później dojść do tablicy routingów, która raczej powinna być prywatna. (Mowa o middleware do autoryzacji).

Reply:
Skoro routing dla Ciebie to nasłuchiwacz połączeń do endpointów, to czemu tworzy połączenie do bazy i tworzy indeksy w bazie? :smile:

Reply:
Dawno troszkę ten fragment kodu pisałem, ale wydaje mi się, że stworzyłem tutaj rzeczy odpowiedzialne za bazę danych, gdyż gdzieś musiałem wstawić wskaźnik na klienta mongo, aby handlery to widziały. Ale faktycznie, mogę to przenieść, bym musiał wtedy stworzyć folder odpowiedzialny za bazę danych i tam w jakiejś metodzie zwrócić wskaźnik? Ale to dalej byłoby rozwiązanie, które niejako inicjuje bazę w routingu, jeśli tam bym to wywołał. Ewentualnie w mainie wywołanie?

Reply:
W tym projekcie to bym to zrobił w mainie. Gdybyś jeszcze zastosował to https://github.com/Daniorocket/cookit-backend/pull/1#discussion_r672092736 to wtedy byś do routingu przekazał tylko te inferejsy a w mainie je inicjował